lib/measured/parser.rb

```diff
@@ -0,0 +1,37 @@
+module Measured::Parser
+  extend self
+
+  PARSE_REGEX = /
+    \A         # beginning of input
+    \s*        # optionally any whitespace
+    (          # capture the value
+      -?       # number can be negative
+      \d+      # must have some digits
+      (?:      # do not capture
+        [\.\/] # period or slash to split fractional part
+        \d+    # some digits after it
+      )?       # fractional part is optional
+    )
+    \s*        # optionally any space between number and unit
+    (          # capture the unit
+      [a-zA-Z] # unit maust start with a letter
+      [\w-]*   # any word characters or dashes
+      (?:      # non capturing group
+        \s+    # there can be a space in the unit
```

Comment:
`can`, or `must`? 

`+` means `must`, since it's one or more, `*` means `can` because it's zero or more.